kratos/geometries/nurbs_surface_geometry.h

```diff
@@ -191,6 +191,22 @@ class NurbsSurfaceGeometry : public Geometry<typename TContainerPointType::value
         return Kratos::make_shared<NurbsSurfaceGeometry>(ThisPoints);
     }
 
+    ///@}
+    ///@name Information
+    ///@{
+
+    /// Returns number of points per direction.
+    SizeType PointsNumberInDirection(IndexType DirectionIndex) const override
+    {
+        if (DirectionIndex == 0) {
+            return this->NumberOfControlPointsU();
+        }
+        else if (DirectionIndex == 1) {
+            return this->NumberOfControlPointsV();
+        }
+        return 1;
```

Comment:
Doesn't it make sense to return an error or a warning, if the directionIndex doesn't match 1 or 2?

Reply:
Thanks for your comment! I am not sure. If you generalize a procedure with loops in all directions it would provide you the appropriate location in the list of nodes. Thus, I think if a user wants to avoid misuse it should ask about the dimensions of the geometries.

Reply:
I think i didn't make myself clear enough. What i'm trying to say is that it might make sense to return a warning or an error instead of returning 1,  if the user input is wrong. For example, you are doing this already for the number of knotspans a couple lines below:

            KRATOS_ERROR << "NurbsSurfaceGeometry::NumberOfKnotSpans: Direction index: "
                << DirectionIndex << " not available. Options are: 0 and 1." << std::endl;

Furthermore i find the name of the function a little confusing. Maybe NumberOfPointsInDirection is a little better, but it's as above just a matter of stlye i guess. 

Cheers

Reply:
thanks @dagmawibekel I try to avoid that warning because what you could do is a generic method that iterates in all directions it iterates only 1 within the thickness direction. This is a legal operation and without lack of efficiency.

I am also not happy with the name, however, the corresponding function is called PointsNumber - that's why I tried to keep that.